src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java

```diff
@@ -55,267 +56,121 @@
-
-            } else if (result.getCode() == ResultCode.CONFLICT) {
-                message = res.getString(R.string.move_file_error);
-            }
-
-        } else if (operation instanceof SynchronizeFolderOperation) {
-
-            if (!result.isSuccess()) {
-                String folderPathName = new File(
-                        ((SynchronizeFolderOperation) operation).getFolderPath()).getName();
-                if (result.getCode() == ResultCode.FILE_NOT_FOUND) {
-                    message = String.format(
-                            res.getString(R.string.sync_current_folder_was_removed),
-                            folderPathName
-                    );
-                }
-            }
-
-        } else if (operation instanceof CopyFileOperation) {
-            if (result.getCode() == ResultCode.FILE_NOT_FOUND) {
-                message = res.getString(R.string.copy_file_not_found);
-
-            } else if (result.getCode() == ResultCode.INVALID_COPY_INTO_DESCENDANT) {
-                message = res.getString(R.string.copy_file_invalid_into_descendent);
-
-            } else if (result.getCode() == ResultCode.INVALID_OVERWRITE) {
-                message = res.getString(R.string.copy_file_invalid_overwrite);
+    public static String getErrorCauseMessage(final RemoteOperationResult result,
+                                           final RemoteOperation operation,
+                                           final Resources resources) {
+
+        final RemoteOperationResult rlt = result;
+        final RemoteOperation o = operation;
+        Formater f = new Formater(resources);
+        if(rlt.isSuccess()) {
+            if(o instanceof UploadFileOperation) return f.f(R.string.uploader_upload_succeeded_content_single, ((UploadFileOperation) o).getFileName());
+            if(o instanceof DownloadFileOperation) return f.f(R.string.downloader_download_succeeded_content);
+            if(o instanceof RemoveFileOperation) return f.f(R.string.remove_success_msg);
+        }
 
-            } else if (result.getCode() == ResultCode.FORBIDDEN) {
-                message = String.format(res.getString(R.string.forbidden_permissions),
-                        res.getString(R.string.forbidden_permissions_copy));
+        if(o instanceof SynchronizeFileOperation &&
+                !((SynchronizeFileOperation) o).transferWasRequested()) {
+            return f.f(R.string.sync_file_nothing_to_do_msg);
+        }
 
-            }
+        if((o instanceof CreateShareViaLinkOperation
+                || o instanceof RemoveShareOperation
+                || o instanceof UpdateShareViaLinkOperation
+                || o instanceof UpdateSharePermissionsOperation)
+                && rlt.getData() != null
+                && rlt.getData().size() > 0) {
+            return rlt.getData().get(0).toString();
         }
 
-        return message;
+        switch (rlt.getCode()) {
+            case LOCAL_STORAGE_FULL:
```

Comment:
It's essentially an or statement within a switch case. It will show the same statement as the line below. However I will fix this for better readability.

Reply:
Do not worry, we are using this way in other parts of the code: https://github.com/owncloud/android/blob/master/src/com/owncloud/android/authentication/AuthenticatorActivity.java#L1290